extensions/wikia/AdEngine/js/provider/gpt/GptHelper.js

```diff
@@ -0,0 +1,90 @@
+/*global define,setTimeout*/
+/*jshint maxlen:125, camelcase:false, maxdepth:7*/
+define('ext.wikia.adEngine.provider.gptHelper', [
+	'wikia.log',
+	'ext.wikia.adEngine.provider.googleTag',
+	'ext.wikia.adEngine.provider.gptAdDetect',
+	'ext.wikia.adEngine.provider.gptAdElement',
+	'ext.wikia.adEngine.slotTweaker',
+	require.optional('ext.wikia.adEngine.provider.gptSraHelper')
+], function (log, googleTag, adDetect, AdElement, slotTweaker, sraHelper) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.provider.gptHelper';
+
+	/**
+	 * Push ad to queue and flush if it should be
+	 *
+	 * @param {string}   slotName           - slot name
+	 * @param {Object}   slotElement        - slot div container
+	 * @param {string}   slotPath           - slot path
+	 * @param {Object}   slotTargeting      - slot targeting details
+	 * @param {Object}   extra              - optional parameters
+	 * @param {function} extra.success      - on success callback
+	 * @param {function} extra.error        - on error callback
+	 * @param {boolean}  extra.sraEnabled   - whether to use Single Request Architecture
+	 * @param {string}   extra.forcedAdType - ad type for callbacks info
+	 */
+	function pushAd(slotName, slotElement, slotPath, slotTargeting, extra) {
+		var element = new AdElement('wikia_gpt_helper' + slotPath);
+
+		extra = extra || {};
+		slotTargeting = JSON.parse(JSON.stringify(slotTargeting)); // copy value
+
+		function callSuccess(adInfo) {
+			if (typeof extra.success === 'function') {
+				extra.success(adInfo);
+			}
+		}
+
+		function callError(adInfo) {
+			slotTweaker.hide(element.getId());
+			if (typeof extra.error === 'function') {
+				adInfo = adInfo || {};
+				adInfo.method = 'hop';
+				extra.error(adInfo);
+			}
+		}
+
+		function queueAd() {
+			log(['queueAd', slotName, slotElement, element], 'debug', logGroup);
+			slotElement.appendChild(element.getNode());
+
+			googleTag.addSlot(slotName, slotPath, slotTargeting, element);
+
+			// Some broken ads never fire "success" event, so we show the div now (and maybe hide later)
+			slotTweaker.show(element.getId());
```

Comment:
I don't think this is needed anymore after the AdEngine refactoring, since we're not hiding the old divs anymore